src/parser/pre2017.py

```diff
@@ -1,7 +1,59 @@
-from typing import Dict
+from typing import Dict, List
 
 from bs4 import BeautifulSoup
 
+CHARGE_SEVERITY = {
+    "First Degree Felony": 1,
+    "Second Degree Felony": 2,
+    "Third Degree Felony": 3,
+    "State Jail Felony": 4,
+    "Misdemeanor A": 5,
+    "Misdemeanor B": 6,
+}
+
+def get_charge_severity(charge: str) -> int:
+    for charge_name, severity in CHARGE_SEVERITY.items():
+        if charge_name in charge:
+            return severity
+    return float('inf')
+
+def count_dismissed_charges(dispositions: List[Dict]) -> int:
+    dismissed_charges = 0
+    for disposition in dispositions:
+        for detail in disposition.get("details", []):
+            if detail.get("outcome", "").lower() == 'dismissed':
+                dismissed_charges += 1
+    return dismissed_charges
+
+def get_top_charge(dispositions: List[Dict], charge_information: List[Dict]) -> Dict:
+    top_charge = None
+    min_severity = float('inf')
+
+    charge_map = {info['charges']: info['level'] for info in charge_information}
+
+    print("Charge Map:", charge_map)
+
+    for disposition in dispositions:
+        for detail in disposition.get("details", []):
+            charge_text = detail.get("charge", "").strip()
+            charge_name = charge_text.split(" >=")[0].strip().lstrip("0123456789. ").strip()
+            charge_level = charge_map.get(charge_name, "Unknown")
+
+            print("Charge Text:", charge_text)
+            print("Charge Name:", charge_name)
+            print("Charge Level:", charge_level)
```

Comment:
We will eventually integrate logging into the code that will replace print statements for testing but we can leave these here to remind us later.